src/snowflake/snowpark/_internal/analyzer/analyzer.py

```diff
@@ -401,6 +402,7 @@ def analyze(
                     expr.api_call_source, TelemetryField.FUNC_CAT_USAGE.value
                 )
             func_name = expr.udf_name.upper() if parse_local_name else expr.udf_name
+            assert expr.children is not None
```

Comment:
Can we eliminate this mypy error by adding a type annotation on SnowflakeUDF.children?